torchdynamo/guards.py

```diff
@@ -52,6 +60,87 @@
 )
 
 
+class NNModuleChangeTrackerUtil:
+    original_register_parameter = None
+    original_add_module = None
+    original_load_state_dict = None
+    original_setattr = None
+
+    @classmethod
+    def register_patches_on_torch_nn_module(cls):
+        if getattr(torch.nn.Module, "_NNModuleChangeTrackerUtil__dynamo_module_patch", True):
```

Comment:
@jansel this was the issue - we were patching over and over and not noticing it, because of a name issue on getattr and hasattr (`_NNModuleChangeTrackerUtil__dynamo_module_patch` vs `__dynamo_module_patch`). We tested on smaller models or use cases where this patch was not getting called as much, and so was still creating the issue, but hiding it. We now only set this up once.